src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -33,7 +33,7 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool up
     if (!i_target.isValid() || !i_target->IsInWorld())
         return;
 
-    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE))
+    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE | UNIT_STATE_CASTING))
```

Comment:
@Keader
here adapted for TC (i think =) 
http://pastebin.com/BDtpbg0V

note: 
SPELL_ATTR5_CAN_CHANNEL_WHEN_MOVING 
use more wide in core,  for example from CMangos: 
https://github.com/cmangos/mangos-wotlk/commit/02796394c1c5d871cc5be1b67b0b415cb78c7ed6

but in R2 more...